falcon/app.py

```diff
@@ -882,7 +897,7 @@ def _prepare_middleware(self, middleware=None, independent_middleware=False):
             middleware=middleware, independent_middleware=independent_middleware
         )
 
-    def _get_responder(self, req):
+    def _get_responder(self, req) -> Tuple[Callable, dict, object, Optional[str]]:
```

Comment:
Should we fully hint this by adding `req: Request`?

Reply:
Yes, I'll add `Request` where applicable. I was trying to use generics for `Request` and `Response` to highlight the fact that one can pass a custom class, but that still seems to have too many rough edges in `mypy`, cannot achieve all I want.